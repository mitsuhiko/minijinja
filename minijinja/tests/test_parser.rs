#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::parse;

#[test]
fn test_parser() {
    insta::glob!("parser-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let filename = path.file_name().unwrap().to_str().unwrap();
        let ast = parse(&contents, filename, Default::default(), Default::default());
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true,
        }, {
            insta::assert_debug_snapshot!(&ast);
        });
    });
}
