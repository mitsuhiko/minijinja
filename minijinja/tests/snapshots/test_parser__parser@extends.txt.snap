---
source: minijinja/tests/test_parser.rs
expression: "&ast"
input_file: minijinja/tests/parser-inputs/extends.txt

---
Ok(
    Template {
        children: [
            Extends {
                name: Const {
                    value: "layout.html",
                } @ 1:11-1:24,
            } @ 1:3-1:27,
            EmitRaw {
                raw: "\n",
            } @ 1:27-2:0,
            Block {
                name: "title",
                body: [
                    EmitRaw {
                        raw: "new title",
                    } @ 2:17-2:26,
                ],
            } @ 2:3-2:40,
            EmitRaw {
                raw: "\n",
            } @ 2:40-3:0,
            Block {
                name: "body",
                body: [
                    EmitRaw {
                        raw: "new body",
                    } @ 3:16-3:24,
                ],
            } @ 3:3-3:38,
        ],
    } @ 1:0-3:38,
)
